schema_based_on_diagram.sql

```diff
@@ -0,0 +1,79 @@
+CREATE TABLE "treatments"(
+    "id" INTEGER NOT NULL,
+    "type" VARCHAR(255) NOT NULL,
+    "name" VARCHAR(255) NOT NULL
+);
+ALTER TABLE
+    "treatments" ADD PRIMARY KEY("id");
+CREATE TABLE "medical_histories"(
+    "id" INTEGER NOT NULL,
+    "admitted_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "patient_id" INTEGER NOT NULL,
+    "status" VARCHAR(255) NOT NULL
+);
+CREATE INDEX "medical_histories_patient_id_index" ON
+    "medical_histories"("patient_id");
+ALTER TABLE
+    "medical_histories" ADD PRIMARY KEY("id");
+CREATE INDEX "medical_histories_patient_id_index" ON
+    "medical_histories"("patient_id");
+CREATE TABLE "Patients"(
+    "id" BIGINT NOT NULL,
+    "name" VARCHAR(255) NOT NULL,
+    "date_of_birth" VARCHAR(255) NOT NULL
+);
+ALTER TABLE
+    "Patients" ADD PRIMARY KEY("id");
+CREATE TABLE "invoice_items"(
+    "id" INTEGER NOT NULL,
+    "unit_price" DECIMAL(8, 2) NOT NULL,
+    "quantity" INTEGER NOT NULL,
+    "total_price" DECIMAL(8, 2) NOT NULL,
+    "invoice_id" INTEGER NOT NULL,
+    "treatment_id" INTEGER NOT NULL
+);
+CREATE INDEX "invoice_items_invoice_id_index" ON
+    "invoice_items"("invoice_id");
+CREATE INDEX "invoice_items_treatment_id_index" ON
+    "invoice_items"("treatment_id");
+ALTER TABLE
+    "invoice_items" ADD PRIMARY KEY("id");
+CREATE INDEX "invoice_items_invoice_id_index" ON
+    "invoice_items"("invoice_id");
+CREATE INDEX "invoice_items_treatment_id_index" ON
+    "invoice_items"("treatment_id");
+CREATE TABLE "medical_treatments"(
+    "medical_histories_id" INTEGER NOT NULL,
+    "treatments_id" INTEGER NOT NULL
+);
+CREATE INDEX "medical_treatments_medical_histories_id_index" ON
+    "medical_treatments"("medical_histories_id");
+CREATE INDEX "medical_treatments_treatments_id_index" ON
+    "medical_treatments"("treatments_id");
```

Comment:
- [ ] The foreign key constraints should be in singular name form so consider changing these two ids in brackets to `medical_history_id` and `treatment_id`